### DISCUSSION TITLE
zh/usage/cli/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#9083] [Docs]
# zh/usage/cli/

了解如何通过命令行使用Ultralytics YOLO  、训练模型、运行预测并使用终端命令轻松将模型导出为不同格式。

https://docs.ultralytics.com/zh/usage/cli/

**Comment by duoerku** [comment]
The Chinese translation of the website is ridiculously wrong. It is recommended to strengthen it, as it is confusing.

**Reply by pderrenger** [reply]
@duoerku 感谢您的反馈！我们确实认识到翻译质量对用户体验至关重要，并非常重视这一点。🌐 我们将努力改进文档的中文翻译，确保其准确性和易于理解。此外，我们欢迎社区成员对改进翻译提出建议或直接贡献。🙌

如果您有具体的改进建议，欢迎通过GitHub issue或pull request与我们分享。我们致力于为所有用户提供高质量的文档。

```bash
# 例如，如果您发现命令行接口(CLI)使用方法部分的某个术语翻译不准确：
yolo train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640
# 我们鼓励提交具体的修正建议！
```

再次感谢您的支持与理解，我们期待您的宝贵建议！ 😊

**Comment by duoerku** [comment]
你好，进一部沟通请联系15680550171， 或者添加微信 BK-Mike

**Reply by pderrenger** [reply]
您好，谢谢您对Ultralytics YOLOv8的兴趣！🚀 对于技术支持和进一步的沟通，请优先利用GitHub Issues功能提出您的问题或需求，这样我们能够提供更有条理且可追踪的支持。

就您提供的信息来看，您似乎对命令行界面（CLI）的使用方法有兴趣，这里是可以通过几个简单示例来帮助您开始：

1. **训练一个模型**：
   ```bash
   yolo train data=coco128.yaml model=yolov8n.pt epochs=10 lr0=0.01
   ```
   
2. **运行预测**：
   ```bash
   yolo predict model=yolov8n-seg.pt source='https://youtu.be/LNwODJXcvt4' imgsz=320
   ```

3. **验证模型**：
   ```bash
   yolo val model=yolov8n.pt data=coco128.yaml batch=1 imgsz=640
   ```

4. **导出模型为ONNX格式**：
   ```bash
   yolo export model=yolov8n-cls.pt format=onnx imgsz=224,128
   ```

更多详细命令和选项，请参阅我们的[官方文档](https://docs.ultralytics.com/zh/usage/cli/)。如果有更具体的问题或需求，欢迎在GitHub上开启一个issue进行讨论。🤗

**Comment by Lishiquan12** [comment]
Usage: yolo [OPTIONS] COMMAND [ARGS]...
Try 'yolo -h' for help.

Error: No such command 'detect'.  why？

**Reply by glenn-jocher** [reply]
嘿！看起来你在尝试使用 `detect` 命令，但这实际上并不直接作为命令使用。😅 在YOLOv8 CLI中，`detect` 更像是一个任务类型标识符，而不是直接使用的命令。

比如说，如果你想要对一张图片进行预测（检测），你可以这样做：

```bash
yolo detect predict model=yolov8n.pt source='path/to/image.jpg'
```

这里 `detect` 指出你想要进行的是物体检测任务，`predict` 是你想要进行的模式，接着是模型和图像源的具体参数。🔍

确保检查 [Ultralytics YOLO 文档](https://docs.ultralytics.com/zh/usage/cli/) 获取更多详细信息和参数选项！希望这有助于解决问题！

**Reply by JackeShen** [reply]
too

**Reply by glenn-jocher** [reply]
看起来你遇到了 `'yolo detect' 是无效命令` 的问题。 😊 对于YOLOv8 CLI的用法, 实际上你不需要直接使用 "detect" 作为命令。以下是一个简单的例子，演示如何正确地使用 CLI 来运行一个预测：

```bash
yolo predict model=yolov8n.pt source='path/to/your/image.jpg'
```
在这个例子中，我们使用 `predict` 而不是 `detect`，然后指定模型和要进行预测的数据源。希望这能帮助到你！如果还有其他疑问，欢迎继续提问。👍

**Comment by duoerku** [comment]
你好，进一部沟通请联系15680550171， 或者添加微信 BK-Mike

**Reply by glenn-jocher** [reply]
@duoerku 你好！很抱歉，我们无法通过电话或微信进行交流，但我们很乐意在这个问题跟踪系统中帮助你解决任何有关Ultralytics YOLO的问题。如果你有任何技术问题或需要进一步的帮助，请随时在这里详细说明，我们会尽快给你回复。感谢你的理解，并期待你的回复。🙂

**Comment by yizhihuo666** [comment]
May I ask which algorithm is used by default for training through the ultralytic command line, YOLOV8 or YOLOV11

**Reply by yizhihuo666** [reply]
I trained yolov8n.pt using the command line in 2023 and recently updated the latest version of the ultralytic library. I found that when I used the original command, it kept prompting me to download the yolo11n.pt file

**Comment by duoerku** [comment]
你好，进一部沟通请联系15680550171， 或者添加微信 BK-Mike

**Reply by glenn-jocher** [reply]
@yizhihuo666 好问题！当你更新到最新版本的Ultralytics库时，默认模型已经从YOLOv8改为YOLO11。这就是为什么你的原始命令现在提示下载yolo11n.pt文件的原因。

如果你想继续使用YOLOv8模型，可以显式指定模型路径：
```bash
yolo train model=yolov8n.pt data=your_data.yaml epochs=100
```

如果你想使用最新的YOLO11模型（推荐），可以直接使用：
```bash
yolo train model=yolo11n.pt data=your_data.yaml epochs=100
```

YOLO11提供了更好的性能和准确性，所以建议升级到新模型。

@duoerku 请在GitHub上提出具体的技术问题，我们会在这里为你提供支持，无法提供私人联系方式。

**Reply by yizhihuo666** [reply]
Thank you for your reply. As you mentioned, the system has now been updated to the YOLOv11 algorithm. I would like to know how I should proceed if I want to train the yolov8n.pt model using the YOLOv8 algorithm. Do I need to revert to a previous version of Ultralytics?

**Reply by glenn-jocher** [reply]
@yizhihuo666 You don't need to downgrade Ultralytics! The current version still fully supports YOLOv8 models. Simply specify the YOLOv8 model explicitly in your command:

```bash
yolo train model=yolov8n.pt data=your_data.yaml epochs=100
```

The latest Ultralytics version supports both YOLOv8 and YOLO11 models - it just defaults to YOLO11 when no model is specified. By explicitly using `model=yolov8n.pt`, you'll train with the YOLOv8 architecture as intended.
